crates/ruff/tests/lint.rs

```diff
@@ -510,6 +510,307 @@ ignore = ["D203", "D212"]
+select = ["I"]
+
+[lint.isort]
+combine-as-imports = true
+        "#,
+    )?;
+    let fixture = r#"
+from foo import (
+    aaaaaaaaaaaaaaaaaaa,
+    bbbbbbbbbbb as bbbbbbbbbbbbbbbb,
+    cccccccccccccccc,
+    ddddddddddd as ddddddddddddd,
+    eeeeeeeeeeeeeee,
+    ffffffffffff as ffffffffffffff,
+    ggggggggggggg,
+    hhhhhhh as hhhhhhhhhhh,
+    iiiiiiiiiiiiii,
+    jjjjjjjjjjjjj as jjjjjj,
+)
+
+x = "longer_than_90_charactersssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss"
+"#;
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(STDIN_BASE_OPTIONS)
+        .arg("--config")
+        .arg(&ruff_toml)
+        .args(["--config", "line-length=90"])
+        .args(["--config", "lint.extend-select=['E501', 'F841']"])
+        .args(["--config", "lint.isort.combine-as-imports = false"])
+        .arg("-")
+        .pass_stdin(fixture), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    -:2:1: I001 [*] Import block is un-sorted or un-formatted
+    -:15:91: E501 Line too long (97 > 90)
+    Found 2 errors.
+    [*] 1 fixable with the `--fix` option.
+
+    ----- stderr -----
+    "###);
+    Ok(())
+}
+
+#[test]
+fn valid_toml_but_nonexistent_option_provided_via_config_argument() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(STDIN_BASE_OPTIONS)
+        .args([".", "--config", "extend-select=['F481']"]),  // No such code as F481!
+        @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: invalid value 'extend-select=['F481']' for '--config <CONFIG_OPTION>'
+
+      tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+      tip: The following error occurred when attempting to parse `extend-select=['F481']` as a `ruff.toml` config option:
+
+    TOML parse error at line 1, column 1
+      |
+    1 | extend-select=['F481']
+      | ^^^^^^^^^^^^^^^^^^^^^^
+    Unknown rule selector: `F481`
+
+    For more information, try '--help'.
+    "###);
+}
+
+#[test]
+fn each_toml_option_requires_a_new_flag_1() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(STDIN_BASE_OPTIONS)
+        // commas can't be used to delimit different config overrides;
+        // you need a new --config flag for each override
+        .args([".", "--config", "extend-select=['F841'], line-length=90"]),
+        @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: invalid value 'extend-select=['F841'], line-length=90' for '--config <CONFIG_OPTION>'
+
+      tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+      tip: The following error occurred when attempting to parse `extend-select=['F841'], line-length=90` as a `ruff.toml` config option:
```

Comment:
I'd make this a little less verbose, maybe?

"Failed to parse `...` as a configuration option"

Additionally, won't the TOML parse error _always_ include the line that the user passed? I guess it's possible that they try to pass multiple lines but that seems very rare. Perhaps including what we tried to parse in the tip message is overkill, and it should just be "It looks like an option was passed, but it was not valid TOML"

Reply:
So the reason I'm specifically saying "`ruff.toml` configuration option" rather than just "configuration option" is because a `pyproject.toml` configuration option would need to be specified differently. E.g. you don't need to specify `--config "tool.ruff.lint.extend-select = ['F841']"` (a la `pyproject.toml`), you just need to specify `--config "lint.extend-select = ['F841']"` (a la `ruff.toml`)